pygeo/constraints/DVCon.py

```diff
@@ -764,6 +764,191 @@ def addThicknessConstraints1D(
             conName, coords, lower, upper, scaled, scale, self.DVGeometries[DVGeoName], addToPyOpt, compNames
         )
 
+    def addProximityConstraints(
+        self,
+        ptList,
+        vecList,
+        surfA,
+        surfB,
+        pointSetKwargsA=None,
+        pointSetKwargsB=None,
+        lower=1.0,
+        upper=3.0,
+        scaled=True,
+        scale=1.0,
+        name=None,
+        addToPyOpt=True,
+        DVGeoName="default",
+        compNames=None,
+    ):
+        r"""
+        Add "toothpick" thickness constraints between multiple surfaces.
+        This can be used to control proximity of multiple components,
+        each defined with its own surface.
+
+        Parameters
+        ----------
+        ptList : list or array of size (N x 3)
+            Together with the vecList, these points will be used to
+            determine the intersections with each component surface.
+
+        vecList : list or array of size (N x 3)
+            Directions that originate from each point in ptList
+            that define the search vector directions for component
+            surface intersections. The vector direction should
+            point to component A, and the opposite direction should
+            point to component B
+
+        surfA : str
+            Name of the surface mesh that defines component A
+
+        surfB : str
+            Name of the surface mesh that defines component B
+
+        pointSetKwargsA : dict
```

Comment:
Does this constraint need pointSetKwargs to work or can the proximity be between the entire surface?